tlatools/org.lamport.tlatools/src/tlc2/model/MCVariable.java

```diff
@@ -1,29 +1,46 @@
 package tlc2.model;
 
 import tlc2.output.EC;
+import tlc2.value.IValue;
 import util.Assert;
 
 public class MCVariable {
     private final String name;
     private final String valueAsString;
```

Comment:
Wouldn't it work to replace the String with IValue instead of keeping both? Or does it fail to invoke IValue#toString at a later stage?

Reply:
At https://github.com/tlaplus/tlaplus/blob/2ef1f6c94dc7eb3ec8ebdaed6190d6a950a0a763/tlatools/org.lamport.tlatools/src/tlc2/model/MCState.java#L368, we create a MCVariable from a disjunction input string, so there is no original TLCValue.

Reply:
Does it hold to assert that IValue and the String are consistent?

Reply:
I can check if they are the same when using the new constructor, is this okay for you? (if tlcValue is not null, of course).

Reply:
Or okay, I can create another constructor like you said which receives only the TCLValue hahahaha, makes more sense.

Reply:
Thanks